config/cmd/cmd.go

```diff
@@ -468,10 +467,79 @@ func (c *cmd) Options() Options {
 }
 
 func (c *cmd) Before(ctx *cli.Context) error {
-	// If flags are set then use them otherwise do nothing
-	var serverOpts []server.Option
+	// Setup client options
 	var clientOpts []client.Option
 
+	if r := ctx.Int("client_retries"); r >= 0 {
+		clientOpts = append(clientOpts, client.Retries(r))
+	}
+
+	if t := ctx.String("client_request_timeout"); len(t) > 0 {
+		d, err := time.ParseDuration(t)
+		if err != nil {
+			logger.Fatalf("failed to parse client_request_timeout: %v", t)
+		}
+		clientOpts = append(clientOpts, client.RequestTimeout(d))
+	}
+
+	if r := ctx.Int("client_pool_size"); r > 0 {
+		clientOpts = append(clientOpts, client.PoolSize(r))
+	}
+
+	if t := ctx.String("client_pool_ttl"); len(t) > 0 {
+		d, err := time.ParseDuration(t)
+		if err != nil {
+			logger.Fatalf("failed to parse client_pool_ttl: %v", t)
+		}
+		clientOpts = append(clientOpts, client.PoolTTL(d))
+	}
+
+	// Setup server options
+	var serverOpts []server.Option
+
+	metadata := make(map[string]string)
+	for _, d := range ctx.StringSlice("server_metadata") {
+		var key, val string
+		parts := strings.Split(d, "=")
+		key = parts[0]
+		if len(parts) > 1 {
+			val = strings.Join(parts[1:], "=")
+		}
+		metadata[key] = val
+	}
+
+	if len(metadata) > 0 {
+		serverOpts = append(serverOpts, server.Metadata(metadata))
+	}
+
+	if len(ctx.String("server_name")) > 0 {
+		serverOpts = append(serverOpts, server.Name(ctx.String("server_name")))
+	}
+
+	if len(ctx.String("server_version")) > 0 {
+		serverOpts = append(serverOpts, server.Version(ctx.String("server_version")))
+	}
+
+	if len(ctx.String("server_id")) > 0 {
+		serverOpts = append(serverOpts, server.Id(ctx.String("server_id")))
+	}
+
+	if len(ctx.String("server_address")) > 0 {
+		serverOpts = append(serverOpts, server.Address(ctx.String("server_address")))
+	}
+
+	if len(ctx.String("server_advertise")) > 0 {
+		serverOpts = append(serverOpts, server.Advertise(ctx.String("server_advertise")))
+	}
+
+	if ttl := time.Duration(ctx.Int("register_ttl")); ttl >= 0 {
```

Comment:
Similar to above `ctx.Int` returns 0 by default so register TTL is set to 0 here if not specified

Reply:
The value is defaulted to 60 when the flag is declared.